### PR TITLE
static-analysis-frontend: Import existing project

### PR DESCRIPTION
**Pull request by La0**
This PR creates a new frontend project, importing [existing code](https://github.com/La0/mozilla-static-analysis), and using the new integration of `yarn2nix`.
It also uses the new sub-project paths

The end goal is to serve this under `static-analysis.moz.tools`

### CODE REVIEW

File: src/staticanalysis/frontend/default.nix [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{ releng_pkgs
+}:
+
+releng_pkgs.lib.mkYarnFrontend {
+  src = ./.;
+	src_path = ./.;
```
Comment (rail) [inline src/staticanalysis/frontend/default.nix]: `src_path` should be "src/staticanalysis/frontend". This is a relative path used to `cd` in some scripts.

Comment (La0) [reply src/staticanalysis/frontend/default.nix]: Unfortunately that does not work, i get:

```
error: getting attributes of path '/app/src/staticanalysis/frontend/src/staticanalysis/frontend': No such file or directory
```

The two options working right now are `./.` or `../../../src/staticanalysis/frontend` (to be explicit)

I guess this issue will be solved when we switch to per-project config

File: lib/please_cli/please_cli/config.py [hunk 2]
```diff
@@ -616,6 +616,64 @@
...
+            ('Checking code quality', 'yarn lint'),
+            ('Running tests', 'yarn test'),
+        ],
+        'deploys': [
+            {
+                'target': 'S3',
+                'options': {
+                    'testing': {
+                        's3_bucket': 'static-analysis-testing-frontend',
+                        'url': 'https://static-analysis.testing.moz.tools',
```
Comment (La0) [inline lib/please_cli/please_cli/config.py]: @garbas I'd like your review on the deploy configuration, especially for the s3_bucket (who creates them ?) and the cloudfront DNS

Comment (garbas) [reply lib/please_cli/please_cli/config.py]: @La0 we are in the process of migrating this responsibility to releng team. so we would be in charge (at least for our stuff) of creating cloudfront DNS. We are currently waiting that everybody returns from PTO. Right now the process is manual, but we will automate (and document it).

Add relevant entries to https://github.com/garbas/build-cloud-tools/blob/master/terraform/base/route53.tf and ping me in the PR. I will merge and deploy it for you.

Comment (garbas) [reply lib/please_cli/please_cli/config.py]: above is for domain stuff.

For S3 bucket + cloudfront we create this manually for now. I hope that once we take over terraform in above, we would also automate this. Do you want me to create this buckets for you?

File: src/staticanalysis/frontend/.neutrinorc.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+const envs = {
+  CONFIG: process.env.CONFIG || 'staging',
+};
+const PORT = process.env.PORT || 8010;
+// HTTPS can be disabled by setting HTTPS_DISABLED environment variable to
+// true. Otherwise it will enforced either using automatically generated
+// certificates or pre-generated ones.
+const HTTPS = process.env.HTTPS_DISABLED ? false :
+  (process.env.SSL_CERT && process.env.SSL_KEY && process.env.SSL_CACERT) ?
```
Comment (garbas) [inline src/staticanalysis/frontend/.neutrinorc.js]: I think we don't set `SSL_CERT`, `SSL_KEY` or `SSL_CACERT`. and new version of `webpack-dev-server` does not include ssl certs anymore. I would like that we refer to our generated ssl certs. Can I push the fix to this branch?

**Comment by La0** [comment]
@garbas Thanks for the review, could you create the 3 S3 buckets and give me their IDs (I don't have access to the Mozilla AWS account). 
I'll update this PR and create the one on `build-cloud-tools` with valid IDs.

File: src/staticanalysis/frontend/.neutrinorc.js (hunk 3 above)
Comment (La0) [reply src/staticanalysis/frontend/.neutrinorc.js]: I just added you as contributor on my repo, you should be able to push.

This code is also used in `shipit-frontend`.

**Comment by garbas** [comment]
@La0 buckets created:
- static-analysis.testing.moz.tools -> d1blqs705aw8h9.cloudfront.net
- static-analysis.staging.moz.tools -> d21hzgxp28m0tc.cloudfront.net
- static-analysis.moz.tools -> d2ezri92497z3m.cloudfront.net

**Comment by La0** [comment]
@garbas I created the PR on [build-cloud-tools](https://github.com/mozilla-releng/build-cloud-tools/pull/351/files)

**Pull request merged**